imageio/imageio-metadata/src/main/java/com/twelvemonkeys/imageio/metadata/tiff/TIFFReader.java

```diff
@@ -135,7 +135,9 @@ protected Directory readDirectory(final ImageInputStream pInput, final long pOff
                 CompoundDirectory next = (CompoundDirectory) readDirectory(pInput, nextOffset, true);
 
                 for (int i = 0; i < next.directoryCount(); i++) {
-                    ifds.add((IFD) next.getDirectory(i));
+                    if(next.getDirectory(i).size() > 0) {
```

Comment:
I'm a little reluctant to this size > 0 thing... 
Can we handle this before even trying `readDirectory`? If not, can we have `readDirectory` return an empty `CompoundDirectory` in this special case where there's really no IFD?